ndbench-core/src/main/java/com/netflix/ndbench/core/NdBenchOperation.java

```diff
@@ -0,0 +1,20 @@
+package com.netflix.ndbench.core;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.google.common.util.concurrent.RateLimiter;
+
+import com.netflix.ndbench.api.plugin.NdBenchMonitor;
+
+public interface NdBenchOperation<K> {
```

Comment:
please add a classdoc. also, is there some functional interface we could make this interface extend? like for example Function or BiFunction?